### PR TITLE
Content uploads

### PR DESCRIPTION
**Pull request by daviddavis**
**Note:** The puppet_utils and rpm_utils files are temporary and the code has been copied directly from pulp so we're not planning to edit the code.

### CODE REVIEW

File: src/katello/client/core/repo.py [hunk 1]
```diff
@@ -32,19 +33,21 @@
...
-
 # base action ----------------------------------------------------------------
 
+
 class RepoAction(BaseAction):
 
     def __init__(self):
         super(RepoAction, self).__init__()
         self.api = RepoAPI()
+        self.upload_api = ContentUploadAPI()
```
Comment (jlsherrill) [inline src/katello/client/core/repo.py:50]: should this be in the base repo action?

File: src/katello/client/lib/puppet_utils.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+def generate_puppet_data(filename):
+    """
+    Extracts the module's metadata file from the tarball. This call will attempt
+    to only extract and read the metadata file itself, cleaning up the
+    extracted file at the end.
+
+    :raise InvalidTarball: if the module file cannot be opened
+    :raise MissingModuleFile: if the module's metadata file cannot be found
+    """
+    temp_dir = os.path.dirname(filename)
```
Comment (jlsherrill) [inline src/katello/client/lib/puppet_utils.py:60]: I'd probably suggest using http://docs.python.org/2/library/tempfile.html  to create the temp_dir

File: src/katello/client/lib/rpm_utils.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    metadata['relativepath'] = os.path.basename(rpm_filename)
+    metadata['filename'] = os.path.basename(rpm_filename)
+
+    # This format is, and has always been, incorrect. As of the new yum importer, the
+    # plugin will generate these from the XML snippet because the API into RPM headers
+    # is atrocious. This is the end game for this functionality anyway, moving all of
+    # that metadata derivation into the plugin, so this is just a first step.
+    # I'm leaving these in and commented to show how not to do it.
+    # metadata['requires'] = [(r,) for r in headers['requires']]
+    # metadata['provides'] = [(p,) for p in headers['provides']]
```
Comment (jlsherrill) [inline src/katello/client/lib/rpm_utils.py:113]: This comment as is probably isn't valid for us, you may want to link to pulp source for our inspiration ?

File: src/katello/client/lib/puppet_utils.py (hunk 2 above)
Comment (daviddavis) [inline src/katello/client/lib/puppet_utils.py:60]: Yea, I fixed this in https://github.com/Katello/katello-cli/pull/82/files#L1R61

File: src/katello/client/lib/rpm_utils.py (hunk 3 above)
Comment (jlsherrill) [inline src/katello/client/lib/rpm_utils.py:113]: didn't read OP, works4me

**Comment by jlsherrill** [comment]
ackity ack ack

File: src/katello/client/lib/rpm_utils.py (hunk 3 above)
Comment (daviddavis) [inline src/katello/client/lib/rpm_utils.py:113]: Talked you on IRC about this but for historical purposes, this comment is from pulp. It was copy-pasted over. I am going to make the comments clearer though.

**Comment by iNecas** [comment]
it walks like an ack, it acks like an ack, it is ack

File: src/katello/client/core/repo.py (hunk 1 above)
Comment (mptap) [inline src/katello/client/core/repo.py:50]: @jlsherrill & @daviddavis : addressing this in a separate PR: https://github.com/Katello/katello-cli/pull/86

**Pull request merged**